csrc/scheduler/utils.cpp

```diff
@@ -252,6 +254,10 @@ size_t mergeNonReduction(TensorView* tv) {
     if (tv->axis(i)->isReduction()) {
       continue;
     }
+    if (tv->axis(i)->isDeviceDim()) {
```

Comment:
`i` is not guaranteed to be the outermost position, right? I suspect you're just assuming that's the case? If so, let's assert that:

```
NVF_ERROR(i == 0, "Invalid device parallelization detected at ", i);
```